packages/components/src/components/Switch/Switch.stories.tsx

```diff
@@ -0,0 +1,91 @@
+import type { ComponentMeta, ComponentStory } from "@storybook/react";
+import { useForm, Controller, SubmitHandler } from "react-hook-form";
+import React from "react";
+import { Box } from "../../primitives/Box";
+import { Button } from "../Button";
+import { Switch } from "./Switch";
+
+export default {
+  component: Switch,
+  title: "Components/Switch",
+} as ComponentMeta<typeof Switch>;
+
+const Template: ComponentStory<typeof Switch> = (args) => <Switch {...args} />;
+
+export const Default = Template.bind({});
+Default.args = {
+  children: "Notifications",
+};
+
+export const Checked = Template.bind({});
+Checked.args = {
+  children: "Enable option",
+  defaultChecked: true,
+  name: "test",
+};
+
+export const Disabled = Template.bind({});
+Disabled.args = {
+  children: "Disabled option",
+  disabled: true,
+};
+
+export const DisabledChecked = Template.bind({});
+DisabledChecked.args = {
+  children: "Disabled option",
+  defaultChecked: true,
+  disabled: true,
+};
+
+export const WithHelpText = Template.bind({});
+WithHelpText.args = {
+  children: "Notifications",
+  helpText: "This enables the notifications settings.",
+};
+
+export const Controlled = (): JSX.Element => {
+  const [checked, setChecked] = React.useState(false);
+
+  return (
+    <Switch
+      checked={checked}
+      onCheckedChange={() =>
+        setChecked((prevIsChecked) => (prevIsChecked === true ? false : true))
+      }
+    >
+      Enable forms
+    </Switch>
+  );
+};
+
+interface FormInputs {
+  enabled: string;
+}
+
+const onSubmit: SubmitHandler<FormInputs> = (data) =>
+  alert(JSON.stringify(data, null, 2));
+
+export const WithReactHookForm = (): JSX.Element => {
+  const { control, handleSubmit } = useForm<FormInputs>({
+    defaultValues: { enabled: "false" },
+  });
+
+  return (
+    <Box.form onSubmit={handleSubmit(onSubmit)}>
+      <Controller
+        control={control}
+        name="enabled"
+        render={({ field }) => (
+          <Switch {...field} onCheckedChange={field.onChange}>
```

Comment:
So default value for the form has the value as a string but wouldn't onCheckedChange change that into a boolean since it uses the boolean `checked` property?

Reply:
I'm seeing this locally for this story 
<img width="541" alt="image" src="https://user-images.githubusercontent.com/3478163/212385707-22f43e1e-eb82-41ff-9300-7eff9a921bfb.png">